File: Cpp/Memory.cpp
// Memory merupakan kelas orang tua kedua dengan urutan
// Produk <- Hardware <- Memory

class Memory : public Hardware
{
	// deklarasi atribut
	private:
		
		int frequency;
		int memorySize;
		string supportCuda;

	// atribut publik
	public:
		// parameter konstruktor
		Memory(int frequency, int memorySize, string supportCuda)
		{
			this->frequency = frequency;
			this->memorySize = memorySize;
			this->supportCuda = supportCuda;
		}
		// Konstruktor
		Memory()
		{
			this->frequency = 0;
			this->memorySize = 0;
			this->supportCuda = "";
		}

		// methods untuk menyimpan frequency
		void setFrequency(int frequency)
		{
			this->frequency = frequency;
		}

		int getFrequency()
		{
			return this->frequency;
		}

		// methods untuk menyimpan ukuran memori
		void setMemorySize(int memorySize)
		{
			this->memorySize = memorySize;
		}

		int getMemorySize()
		{
			return this->memorySize;
		}

		// methods untuk menyimpan indikator kuda
		void setSupportCuda(string supportCuda)
		{
			this->supportCuda = supportCuda;
		}

		string getSupportCuda()
		{
			return this->supportCuda;
		}

		// Destruktor untuk membersihkan memory
		~Memory(){

		}

};

File: PHP/Memory.php
<?php 

    // Memory merupakan kelas orang tua kedua dengan urutan
    // Produk <- Hardware <- Memory

    class Memory extends Hardware
    {
        // atribut publik
        public $frequency;
        public $memorysize;
        public $cuda;

        // Konstruktor
        public function __construct($frequency = "", $memorysize = "", $cuda = "")
        {
            $this->frequency = $frequency;
            $this->memorysize = $memorysize;
            $this->cuda = $cuda;            
        }

        // methods untuk menyimpan frequency
        public function setFrequency($frequency)
        {
            $this->frequency = $frequency;
        }   
        public function getFrequency()
        {
            return $this->frequency;
        }

        // methods untuk menyimpan ukuran memori
        public function setMemorySize($memorysize)
        {
            $this->memorysize = $memorysize;
        }   
        public function getMemorySize()
        {
            return $this->memorysize;
        }
        
        // methods untuk menyimpan indikator kuda
        public function setCuda($cuda)
        {
            $this->cuda = $cuda;
        }   
        public function getCuda()
        {
            return $this->cuda;
        }

        // Destruktor untuk membersihkan memory
        public function __destruct()
        {
            
        }
    }
?>